pkg/controllers/nodeclass/status/images.go

```diff
@@ -112,6 +113,16 @@ func (r *NodeImageReconciler) Reconcile(ctx context.Context, nodeClass *v1beta1.
 	ctx = log.IntoContext(ctx, log.FromContext(ctx).WithName(nodeImageReconcilerName))
 	logger := log.FromContext(ctx)
 
+	// validate FIPS + useSIG
+	fipsMode := nodeClass.Spec.FIPSMode
+	useSIG := options.FromContext(ctx).UseSIG
+	if fipsMode != nil && *fipsMode == v1beta1.FIPSEnabled && !useSIG {
+		nodeClass.Status.Images = nil
+		nodeClass.StatusConditions().SetFalse(v1beta1.ConditionTypeImagesReady, "SIGRequiredForFIPS", "FIPS images require UseSIG to be enabled, but UseSIG is false")
```

Comment:
Not recommending any update, here, but considering if it makes sense to mention being unsupported for `self-hosted` here. However, that's implied logic, not direct.

Curious other reviewers opinions.

Reply:
Discussed with Matthew and Alex. Let's add some context for where use is allowed.

```suggestion
		nodeClass.StatusConditions().SetFalse(v1beta1.ConditionTypeImagesReady, "SIGRequiredForFIPS", "FIPS images require UseSIG to be enabled, but UseSIG is false (note: UseSIG is only supported in AKS managed NAP)")
```

Reply:
Also, update the log message